### PR TITLE
Add GH release step on CI and tweak changelog styles

### LINKED ISSUE #26: Change log should be in Keep A Changlog format
Otherwise the automation for creating GitHub releases doesn't copy the contents over.

### PR DESCRIPTION
**Pull request by Goooler**
Closes #26.

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -25,9 +26,13 @@ jobs:
...
-      - uses: ffurrer2/extract-release-notes@v2
-        id: release_notes
-
-      - uses: ncipollo/release-action@v1
+      - name: Extract release notes
+        id: extract-release-notes
+        uses: ffurrer2/extract-release-notes@v2
         with:
-          body: ${{ steps.release_notes.outputs.release_notes }}
+          release_notes_file: RELEASE_NOTES.md
```
Comment (JakeWharton) [inline .github/workflows/release.yaml:33]: This is not the name of our file. We can rely on the default

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -25,9 +26,13 @@ jobs:
...
+
+      - name: Create GitHub release
+        run: gh release create ${{ github.ref_name }} --notes-file RELEASE_NOTES.md
```
Comment (JakeWharton) [inline .github/workflows/release.yaml:36]: Same here

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,24 +2,35 @@
...
 
 
-## Version 0.1.0 *(2024-10-08)*
+## [0.1.0] *(2024-10-08)*
 
 Initial release. We're rebooting the [Burst] project that released 1.0 ten years ago.
 
 [Burst]: https://github.com/square/burst
+[Unreleased]: https://github.com/cashapp/burst/compare/0.4.0...HEAD
+[0.4.0]: https://github.com/cashapp/burst/releases/tag/0.4.0
```
Comment (JakeWharton) [inline CHANGELOG.md:33]: Please put the links directly below the headings

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,24 +2,35 @@
 
 ## Unreleased
```
Comment (JakeWharton) [inline CHANGELOG.md:3]: This is missing square brackets, and please move the link up to be on the next line.

File: .github/workflows/release.yaml (hunk 1 above)
Comment (Goooler) [reply .github/workflows/release.yaml:33]: It's used by `--notes-file` option, due to https://github.com/cashapp/licensee/pull/357#issue-2506630710.

Comment (JakeWharton) [reply .github/workflows/release.yaml:33]: Oh this _writes_ the file, not reads it.

File: CHANGELOG.md (hunk 3 above)
Comment (Goooler) [reply CHANGELOG.md:43]: Updated. GH didn't show the updated state :)

**Pull request merged**